tests/scalers/datadog.test.ts

```diff
@@ -0,0 +1,286 @@
+environment variables
+
+DATADOG_API_KEY
+DATADOG_APP_KEY
+DATADOG_SITE (optional, default datadoghq.com)
+
+*/
+import * as fs from 'fs'
+import * as sh from 'shelljs'
+import * as tmp from 'tmp'
+import test from 'ava'
+
+const datadogApiKey = process.env['DATADOG_API_KEY']
+const datadogAppKey = process.env['DATADOG_APP_KEY']
+let datadogSite = process.env['DATADOG_SITE']
+const testNamespace = 'datadog-test'
+const datadogNamespace = 'datadog'
+const datadogHelmRepo = 'https://helm.datadoghq.com'
+const datadogHelmRelease = 'datadogkeda'
+const kuberneteClusterName = 'keda-datadog-cluster'
+
+test.before(t => {
+  if (!datadogApiKey) {
+    t.fail('DATADOG_API_KEY environment variable is required for Datadog tests')
+  }
+  if (!datadogAppKey) {
+    t.fail('DATADOG_APP_KEY environment variable is required for Datadog tests')
+  }
+  if (!datadogSite) {
+    datadogSite = 'datadoghq.com'
+  }
+
+  sh.exec(`helm repo add datadog ${datadogHelmRepo}`)
+  sh.exec(`helm repo update`)
+  let helmInstallStatus = sh.exec(`helm upgrade \
+  		--install \
+  		--set datadog.apiKey=${datadogApiKey} \
+  		--set datadog.appKey=${datadogAppKey} \
+		  --set datadog.site=${datadogSite} \
+		  --set datadog.clusterName=${kuberneteClusterName} \
+		  --set datadog.kubelet.tlsVerify=false \
+		  --create-namespace \
+		  --namespace ${datadogNamespace} \
+        ${datadogHelmRelease} datadog/datadog`).code
+  sh.echo(`${helmInstallStatus}`)
+  t.is(0,
+    helmInstallStatus,
+    'deploying the Datadog Helm chart should work.'
+  )
+
+  sh.config.silent = true
+
+  // Let's wait until the Datadog agent is ready
+  let datadogDesired = sh.exec(`kubectl get daemonset ${datadogHelmRelease} \
+    --namespace ${datadogNamespace} -o jsonpath="{.status.desiredNumberScheduled}"`).stdout
+
+  let datadogReady = sh.exec(`kubectl get daemonset ${datadogHelmRelease} \
+    --namespace ${datadogNamespace} -o jsonpath="{.status.numberReady}"`).stdout
+
+  while (datadogReady != datadogDesired) {
+    sh.exec('sleep 2')
+    datadogDesired = sh.exec(`kubectl get daemonset ${datadogHelmRelease} \
+      --namespace ${datadogNamespace} -o jsonpath="{.status.desiredNumberScheduled}"`).stdout
+
+    datadogReady = sh.exec(`kubectl get daemonset ${datadogHelmRelease} \
+      --namespace ${datadogNamespace} -o jsonpath="{.status.numberReady}"`).stdout
+  }
+
+  const tmpFile = tmp.fileSync()
+  fs.writeFileSync(tmpFile.name, deployYaml)
+  sh.exec(`kubectl create namespace ${testNamespace}`)
+  t.is(
+    0,
+    sh.exec(`kubectl apply -f ${tmpFile.name} --namespace ${testNamespace}`).code,
+    'creating a deployment should work.'
+  )
+
+  t.is(
+    0,
+    sh.exec(`kubectl create secret generic datadog-secrets --from-literal=apiKey=${datadogApiKey} \
+      --from-literal=appKey=${datadogAppKey} --from-literal=datadogSite=${datadogSite} --namespace ${testNamespace}`).code,
+    'creating a generic secret should work.'
+  )
+
```

Comment:
Please, create first the secret and then the scaled object, there is a known bug we are working on that produces errors if the secrets are not available during the SO creation. In this case the SO is created before the secret and that why sometimes it works and other times not

```suggestion
  t.is(
    0,
    sh.exec(`kubectl create secret generic datadog-secrets --from-literal=apiKey=${datadogApiKey} \
      --from-literal=appKey=${datadogAppKey} --from-literal=datadogSite=${datadogSite} --namespace ${testNamespace}`).code,
    'creating a generic secret should work.'
  )

  const tmpFile = tmp.fileSync()
  fs.writeFileSync(tmpFile.name, deployYaml)
  sh.exec(`kubectl create namespace ${testNamespace}`)
  t.is(
    0,
    sh.exec(`kubectl apply -f ${tmpFile.name} --namespace ${testNamespace}`).code,
    'creating a deployment should work.'
  )

```

Reply:
I have now changed the order as you suggested, and I also took the opportunity to improve the output messages in case of test failure, so we can understand it better.